kratos/utilities/sub_model_part_operations_utility.h

```diff
@@ -0,0 +1,165 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main authors:    Miguel Maso Sotomayor
+//
+
+#ifndef KRATOS_SUB_MODEL_PART_OPERATIONS_H_INCLUDED
+#define KRATOS_SUB_MODEL_PART_OPERATIONS_H_INCLUDED
+
+
+// External includes
+
+
+// Project includes
+#include "includes/model_part.h"
+#include "processes/entity_erase_process.h"
+
+
+namespace Kratos
+{
+///@addtogroup KratosCore
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/**
+ * @class SubModelPartOperationsUtility
+ * @brief Wrapper of set operations: union, intersection and difference
+ * @author Miguel Maso Sotomayor
+ * @ingroup KratosCore
+*/
+class KRATOS_API(KRATOS_CORE) SubModelPartOperationsUtility
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    /// Pointer definition of SubModelPartOperationsUtility
+    KRATOS_CLASS_POINTER_DEFINITION(SubModelPartOperationsUtility);
+
+    typedef std::size_t IndexType;
+
+    ///@}
+    ///@name  Enum's
+    ///@{
+
+    enum class SetOperators {Union, Intersection, Difference};
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    /// Destructor.
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    template<class TEntityType, class TContainerType>
+    void Union(ModelPart& rModelPartA, ModelPart& rModelPartB, ModelPart& rDestination)
```

Comment:
why not `static`?

Reply:
I'm going to make it static

Reply:
Now it is static. The python process is slightly different since the class is not constructed